### ISSUE TITLE
expecting a string with a ${{..}} expression or matrix error

### ISSUE THREAD

**Issue by pcassaretto-cmc** [rhysd/actionlint#205]
When using the following expression
```
strategy:
      matrix: ${{ fromJson(format('{{ "include"{0} [{{ "project"{0} "{1}" }}] }}',':',github.event.inputs.product)) }}
```
get the following error 

`expecting a string with ${{...}} expression or matrix, but found plain text node [syntax-check]`

The expression works fine and will create a job instance for the single item?

Using actionlint via https://github.com/reviewdog/action-actionlint using release `v.1.28.0` which uses actionlint `1.6.16`

**Comment by pcassaretto-cmc** [comment]
Here is an action that can be manually invoked which will reproduce the issue

```
name: test manual invocation
on:
  workflow_dispatch:
    inputs:
      product:
        description: "The product for which to run this workflow"
        required: true

jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      matrix: ${{ fromJson(format('{{ "include"{0} [{{ "project"{0} "{1}" }}] }}',':',github.event.inputs.product)) }}

    steps:
      - uses: actions/checkout@v3.0.2
      - run: |
          echo the project value is ${{ matrix.project }}
```

**Comment by rhysd** [comment]
This was fixed in [v1.6.17](https://github.com/rhysd/actionlint/releases/tag/v1.6.17).
